core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -886,4 +814,34 @@ private void registerQuery(QuerySpec<?> spec) {
         queryParserRegistry.register(spec.getParser(), spec.getName());
         namedWriteables.add(new Entry(QueryBuilder.class, spec.getName().getPreferredName(), spec.getReader()));
     }
+
+    /**
+     * Package private with the intent that it is used for testing.
+     */
+    interface TestPlugin extends SearchPlugin {
```

Comment:
I don't think we need test code in production. Instead, can you add this method to Node, and have MockNode override it? Eventually, we can change it to `createSearchService`, once we can construct `IndicesService` without guice (really close..we can probably do that in a follow up, all the pieces are attainable).